utils/argent-context.js

```diff
@@ -0,0 +1,143 @@
+/* global artifacts */
+
+const ethers = require("ethers");
+const BN = require("bn.js");
+
+const utils = require("./utilities.js");
+const RelayManager = require("./relay-manager");
+
+const WalletFactory = artifacts.require("WalletFactory");
+const BaseWallet = artifacts.require("BaseWallet");
+const ModuleRegistry = artifacts.require("ModuleRegistry");
+const TransferStorage = artifacts.require("TransferStorage");
+const GuardianStorage = artifacts.require("GuardianStorage");
+const ArgentModule = artifacts.require("ArgentModule");
+const DappRegistry = artifacts.require("DappRegistry");
+const ERC20 = artifacts.require("TestERC20");
+
+const { ETH_TOKEN } = utils;
+const ZERO_ADDRESS = ethers.constants.AddressZero;
+const SECURITY_PERIOD = 2;
+const SECURITY_WINDOW = 2;
+const RECOVERY_PERIOD = 4;
+const LOCK_PERIOD = 4;
+
+let tokensInitialized = false;
+
+class ArgentContext {
+  constructor(accounts) {
+    [
+      this.infrastructure,
+      this.owner,
+      this.guardian1,
+      , // eslint-disable-line
+      this.relayer, // 4
+      , // eslint-disable-line
+      this.tokensAddress, // 6
+      this.refundAddress, // 7
+    ] = accounts;
+  }
+
+  async initialize() {
+    const moduleRegistry = await ModuleRegistry.new();
+    const guardianStorage = await GuardianStorage.new();
+    const transferStorage = await TransferStorage.new();
+    this.dappRegistry = await DappRegistry.new(0);
+    const uniswapRouter = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
+    this.module = await ArgentModule.new(
+      moduleRegistry.address,
+      guardianStorage.address,
+      transferStorage.address,
+      this.dappRegistry.address,
+      uniswapRouter,
+      SECURITY_PERIOD,
+      SECURITY_WINDOW,
+      RECOVERY_PERIOD,
+      LOCK_PERIOD,
+    );
+    await moduleRegistry.registerModule(this.module.address, ethers.utils.formatBytes32String("ArgentModule"));
+    await this.dappRegistry.addDapp(0, this.relayer, ZERO_ADDRESS);
+
+    const walletImplementation = await BaseWallet.new();
+    this.factory = await WalletFactory.new(
+      walletImplementation.address,
+      guardianStorage.address,
+      this.refundAddress,
+    );
+    await this.factory.addManager(this.infrastructure);
+    this.manager = new RelayManager(guardianStorage.address, ZERO_ADDRESS);
+
+    await this.initializeTokens();
+    return this;
+  }
+
+  // transfer tokens from mainnet whale addresses to a test address we control
+  async initializeTokens() {
+    this.DAI = await ERC20.at("0x6b175474e89094c44da98b954eedeac495271d0f");
+    this.WETH = await ERC20.at("0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2");
+    this.USDC = await ERC20.at("0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"); 
+
+    if (tokensInitialized) {
```

Comment:
shouldn't this be at the beginning of the function?

Reply:
here we want to do the transfer only once, but still instantiate the tokens for every test file. I've renamed the variable for more clarity